pkg/apis/policy/types.go

```diff
@@ -232,6 +232,10 @@ type PodSecurityPolicySpec struct {
 	// Empty or nil indicates that only the DefaultProcMountType may be used.
 	// +optional
 	AllowedProcMountTypes []api.ProcMountType
+	// AllowedCSIDrivers is a whitelist of allowed CSI drivers.  This slice will be checked
```

Comment:
Should this clarify that it's only limiting inline csi drivers (and not pv.csi drivers)?

Reply:
This will be a way to whitelist any CSI driver, not just inline.

Reply:
If that's the case, then I think we may need two separate fields, one for inline and one for PV.  For example, we want to allow gce PD CSI driver as a PV, but disallow it as inline because a lot of functionality won't work inline.

Reply:
whether it will work or not is not the point of PSP. you can specify driver names that don't even exist in a CSI PV today, and it fails at the point of calling the driver, correct?

PSP is about explicitly whitelisting which drivers a user is allowed to reference

Reply:
@msau42 to reduce complexity, the meaning of the PSP is simply a list of allowed CSI drivers here. Capabilities such as inline/attached etc is exposed in the `CSIInfo` object.  We can refine its interpretation in upcoming release.

Reply:
Is PSP validated at admission time and/or runtime?  I thought that today we cannot have PV validation in PSP because of cross-object validation and the PV may not exist yet

Reply:
Admission time. PSP only applies to pods, so this would only restrict inline CSI volume sources